identity-wallet/src/state/reducers/dev_mode.rs

```diff
@@ -38,20 +42,248 @@ lazy_static! {
+        .post(url)
+        .json(&payload)
+        .send()
+        .await
+        .map_err(|err| AppError::DevError(err.to_string()))?
+        .json()
+        .await
+        .map_err(|err| AppError::DevError(err.to_string()))?;
+
+    let qr_code = QrCodeScanned {
+        form_urlencoded: response.uri,
+    };
+
+    let state = command::reduce(state, Arc::new(qr_code)).await?;
+
+    command::reduce(state, Arc::new(ConnectionAccepted)).await
+}
 
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+struct PresentationResponse {
+    uri: String,
+    request: serde_json::Value,
+    #[serde(rename = "requestOptions")]
+    request_options: serde_json::Value,
+}
+
+async fn add_presentation_request(state: AppState) -> Result<AppState, AppError> {
+    // URL from NGDIL demo
+    let url = "https://api.demo.ngdil.com/api/oid4vp";
+
+    let payload = serde_json::json!({
+        "presentationStage":"dominiqueEnrolCourse",
+        "clientMetadata":{"logoUri":"https://demo.ngdil.com/imgs/kw1c-white.png","clientName":"Koning Willem I College"}
+    });
+
+    let response: PresentationResponse = reqwest::Client::new()
+        .post(url)
+        .json(&payload)
+        .send()
+        .await
+        .map_err(|err| AppError::DevError(err.to_string()))?
+        .json()
+        .await
+        .map_err(|err| AppError::DevError(err.to_string()))?;
+
+    let qr_code = QrCodeScanned {
+        form_urlencoded: response.uri,
+    };
+
+    let state = command::reduce(state, Arc::new(qr_code)).await?;
+
+    if let Some(CurrentUserPrompt::ShareCredentials {
+        client_name: _,
+        logo_uri: _,
+        options,
+    }) = &state.current_user_prompt
+    {
+        let credential_uuids: Vec<Uuid> = options
+            .iter()
+            .map(|uuid_str| Uuid::parse_str(uuid_str).unwrap())
+            .collect();
+
+        let cr_selected = CredentialsSelected { credential_uuids };
+
+        return command::reduce(state, Arc::new(cr_selected)).await;
+    }
+
+    Err(AppError::DevError("Presentation not accepted".to_string()))
+}
+
+async fn add_future_engineer(state: AppState) -> Result<AppState, AppError> {
+    let url = "https://api.demo.ngdil.com/api/credential-offer";
+
+    let payload = json!({"credential":"Future Engineer","issuer":"kw1c"});
+
+    let response: CredentialResponse = reqwest::Client::new()
+        .post(url)
+        .json(&payload)
+        .send()
+        .await
+        .map_err(|err| AppError::DevError(err.to_string()))?
+        .json()
+        .await
+        .map_err(|err| AppError::DevError(err.to_string()))?;
+
+    let qr_code = QrCodeScanned {
+        form_urlencoded: response.uri,
+    };
+
+    let state = command::reduce(state, Arc::new(qr_code)).await?;
+
+    let cr_selected = CredentialOffersSelected {
+        offer_indices: vec![0],
+    };
+
+    command::reduce(state, Arc::new(cr_selected)).await
+}
+
+async fn reset_settings(state: AppState) -> Result<AppState, AppError> {
```

Comment:
What "settings" are reset here? It just reduces the `reset_state()` under the hood, right? Why is this extra function needed?

Reply:
I just made for every step a function to keep it consistent, its also easy to read